group_project_v2/group_project.py

```diff
@@ -446,12 +445,23 @@ def grade_display_stages(self):
         return self.get_children_by_category(GradeDisplayStage.CATEGORY)
 
     @property
-    def max_grade_display_date(self):
+    def get_grade_display_stage(self):
         """
-        Gets max grade display date.
+        Gets grade display stage with max open date.
         """
-        stage_open_dates = [stage.open_date for stage in self.grade_display_stages if stage.open_date]
-        return max(stage_open_dates) if stage_open_dates else None
+        stages = self.grade_display_stages
+        if not stages:
+            return None
+
+        grade_display_stage = stages[0]
+        if len(stages) == 1:
+            return grade_display_stage
+
+        for stage in stages[1:]:
+            if stage.open_date and stage.open_date > grade_display_stage.open_date:
+                grade_display_stage = stage
```

Comment:
@ziafazal looks like it is equivalent to

    if stages:
        grade_display_stage = max(stages, key = lambda stage: stage.open_date if stage.open_date else date.min)
    else:
        grade_display_stage = None